recipes/valijson/all/conanfile.py

```diff
@@ -0,0 +1,34 @@
+from conans import ConanFile, tools
+import os
+import glob
+
+
+class ValijsonConan(ConanFile):
+    name = "valijson"
+    description = "Valijson is a header-only JSON Schema Validation library for C++11."
+    topics = ("conan", "valijson", "json", "validator")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/tristanpenman/valijson"
+    license = "BSD-2-Clause"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = glob.glob(self.name + "-*/")[0]
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def package(self):
+        include_folder = os.path.join(self._source_subfolder, "include")
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*", dst="include", src=include_folder)
+
+    def package_id(self):
+        self.info.header_only()
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "valijson"
+        self.cpp_info.names["cmake_find_package_multi"] = "valijson"
```

Comment:
This does not look like it coversthe CMake targets

https://github.com/tristanpenman/valijson#how-to-add-this-library-to-your-cmake-target

> target_link_libraries(your-executable ValiJSON::valijson)

Reply:
Oh, my wrong.
Could you please help me, how should I change it?

Reply:
Space is our expert, here's an example that recently got merged, https://github.com/conan-io/conan-center-index/pull/2559/files#diff-559d1628b195b97e815856556814668fR210-R226

If you need more help, ill do my best

Reply:
by default it uses the package name, so you don't need it.

Reply:
Sorry, @prince-chrismc and @uilianries .
I don't understand how to fix the issue.

Should I rename my package?
Or should I somehow add components? There's only one component, I suppose, but I don't understand where ValiJSON::valijson should go.

Reply:
This should do the trick

```suggestion
    def package_info(self):
        
        # self.cpp_info.filenames["cmake_find_package"] = "valijson" # TBA: There's no installed config file
        # self.cpp_info.filenames["cmake_find_package_multi"] = "valijson" # TBA: There's no installed config file
        self.cpp_info.names["cmake_find_package"] = "ValiJSON"
        self.cpp_info.names["cmake_find_package_multi"] = "ValiJSON"

        self.cpp_info.components["libvalijson"].names["cmake_find_package"] = "valijson"
        self.cpp_info.components["libvalijson"].names["cmake_find_package_multi"] = "valijson"
        self.cpp_info.components["libvalijson"].libs = [self._lib_name("valijson")]
```

Reply:
"libvalijson" - is it some semi-random string? For example, what will change, if I name it lib_valijson or something else?

self._lib_name is missing, I think in this case it's just valijson